tenseal/tensealcontext.cpp

```diff
@@ -240,6 +254,14 @@ bool TenSEALContext::auto_mod_switch() {
     return this->_auto_flags & flag_auto_mod_switch;
 }
 
+bool TenSEALContext::equals(const std::shared_ptr<TenSEALContext>& other) {
+    if (this->parms() != other->parms()) return false;
+
+    if (this->safe_global_scale() != other->safe_global_scale()) return false;
+
+    return true;
+}
+
```

Comment:
I'm not sure if this the right way to compare two TenSEALContext, since it's have a SEALContext inside.
SEAL will check internally the validation of context and encryption parameters while doing arithmetic operations as well as serialization.
I don't think we are going to compare two TenSEALContext at all, is there any use case for that?
ping @youben11

Reply:
I'm actually not sure at all about that, as Bogdan pointed out in a previous comment, what about the other keys?

Also, if the scale isn't the same, operations aren't well defined, so we can't only leave it to SEAL to make the check. This needs more analysis actually and I would prefer to move it to another issue/PR to work on it specifically.

Reply:
I'll revert it to the previous pointer comparison in this PR.

Reply:
This will make it not work for our use case, I think you can leave it as is with equals method, but make equals always return True with a TODO comment

Reply:
Yeah, you are right
I forget about the scale, if it's different it will lead to wrong result.

Reply:
https://github.com/OpenMined/TenSEAL/issues/112